components/treeview/data-binding/load-on-demand.md

```diff
@@ -18,9 +18,108 @@ You don't have to provide all the data the treeview will render at once - the ro
 
 In this article:
 
+* [Hierarchical Data Load on Demand - One Model](#hierarchical-data-load-on-demand---one-model)
+
 * [Flat Data Load on Demand](#flat-data-load-on-demand)
 
-* [Hierarchical Data Load on Demand](#hierarchical-data-load-on-demand)
+* [Hierarchical Data Load on Demand - Different Models](#hierarchical-data-load-on-demand---different-models)
+
+## Hierarchical Data Load on Demand - One Model
+
+The **example** below shows how you can handle hierarchical data load on demand in detail. It uses the same model for the two different [levels of data bindings]({%slug components/treeview/data-binding/overview%}#multiple-level-bindings) it showcases.
+
+>caption One Model Hierarchical Data Load on Demand in a TreeView with sample handling of the various cases. Review the code comments for details.
+
+````CSHTML
+@using Telerik.DataSource.Extensions
+@* used for the .AddRange() extension method *@
+
+<TelerikTreeView Data="@HierarchicalData" OnExpand="@LoadChildren">
+    <TreeViewBindings>
+        <TreeViewBinding TextField="ProductName" ItemsField="SubProducts" />
+    </TreeViewBindings>
+</TelerikTreeView>
+
+@code {
+    public List<ProductItem> HierarchicalData { get; set; }
+
+    public class ProductItem
+    {
+        public string ProductName { get; set; }
+        public int Id { get; set; } //will be used to identify the node, not for rendering in this example
+        public List<ProductItem> SubProducts { get; set; }
+        public bool Expanded { get; set; }
+        public bool HasChildren { get; set; }
```

Comment:
Consider defaulting this field to true in the model, so you don't have to initialize it in the sample data. I imagine that would be more common so you don't have to check whether the item has children when loading it.